packages/google_sign_in/google_sign_in_platform_interface/lib/src/method_channel_google_sign_in.dart

```diff
@@ -33,27 +34,27 @@ class MethodChannelGoogleSignIn extends GoogleSignInPlatform {
   }
 
   @override
-  Future<GoogleSignInUserData> signInSilently() {
+  Future<GoogleSignInUserData?> signInSilently() {
     return channel
         .invokeMapMethod<String, dynamic>('signInSilently')
         .then(getUserDataFromMap);
   }
 
   @override
-  Future<GoogleSignInUserData> signIn() {
+  Future<GoogleSignInUserData?> signIn() {
     return channel
         .invokeMapMethod<String, dynamic>('signIn')
         .then(getUserDataFromMap);
   }
 
   @override
   Future<GoogleSignInTokenData> getTokens(
-      {String email, bool shouldRecoverAuth = true}) {
+      {required String email, bool? shouldRecoverAuth = true}) {
```

Comment:
`shouldRecoverAuth` has a default,  can it also accept `null`?

Reply:
The interface does not define a default. Should it?

```dart
Future<GoogleSignInTokenData> getTokens(
      {required String email, bool? shouldRecoverAuth}) async {
    throw UnimplementedError('getTokens() has not been implemented.');
  }
```

Reply:
Got it.  It looks like `null` is trying to workaround the fact that not all implementations of the interface need  this field.